src/runtime/metal.cpp

```diff
@@ -700,10 +701,26 @@ WEAK int halide_metal_run(void *user_context,
             }
         }
 
-        if (total_args_size < 4096 && metal_api_supports_set_bytes) {
+        // The Metal compiler introduces padding of some unknown size to the end
+        // of the struct in some (apparently undocumented) circumstances.  We
+        // replicate the logic of this gist:
+        // https://gist.github.com/btipling/fa0ab7bd514d8a319c886def0980ccfc
+        // and pad to a multiple of the largest argument size
+        //
+        // Alternatively, we could use reflection when creating the pipeline state object
+        // and then check the first MTLArgument to determine the size.
+        //
+        // TODO(shoaibkamil): Follow up with Apple to find a specification for when
+        // and how much padding is needed
+        size_t padded_args_size = (total_args_size + arg_sizes[0] - 1) & -arg_sizes[0];
```

Comment:
Is & -arg_sizes[0] a common idiom? I had to think way too hard about that. I think elsewhere in the compiler we do & ~(arg_sizes[0] - 1)